contracts/core/OTCWrapper.sol

```diff
@@ -0,0 +1,690 @@
+    mapping(address => bool) public isWhitelistedMarketMaker;
+
+    ///@notice mapping between an asset address and its corresponding fee
+    mapping(address => uint256) public fee;
+
+    /************************************************
+     *  CONSTRUCTOR & INITIALIZATION
+     ***********************************************/
+
+    /**
+     * @notice constructor related to ERC2771
+     * @param _trustedForwarder trusted forwarder address
+     */
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor(MinimalForwarder _trustedForwarder) ERC2771ContextUpgradeable(address(_trustedForwarder)) {}
+
+    /**
+     * @notice initialize the deployed contract
+     * @param _beneficiary beneficiary address
+     * @param _addressBook AddressBook address
+     * @param _owner owner address
+     * @param _fillDeadline fill deadline duration
+     * @param _usdc USDC address
+     */
+    function initialize(
+        address _addressBook,
+        address _owner,
+        address _beneficiary,
+        uint256 _fillDeadline,
+        address _usdc
+    ) external initializer {
+        require(_addressBook != address(0), "OTCWrapper: addressbook address cannot be 0");
+        require(_owner != address(0), "OTCWrapper: owner address cannot be 0");
+        require(_beneficiary != address(0), "OTCWrapper: beneficiary address cannot be 0");
+        require(_fillDeadline > 0, "OTCWrapper: fill deadline cannot be 0");
+        require(_usdc != address(0), "OTCWrapper: usdc address cannot be 0");
+
+        __Ownable_init();
+        __ReentrancyGuard_init();
+
+        addressbook = AddressBookWrapperInterface(_addressBook);
+        marginRequirements = MarginRequirementsWrapperInterface(addressbook.getMarginRequirements());
+        controller = ControllerWrapperInterface(addressbook.getController());
+        oracle = OracleWrapperInterface(addressbook.getOracle());
+        whitelist = WhitelistWrapperInterface(addressbook.getWhitelist());
+
+        beneficiary = _beneficiary;
+        fillDeadline = _fillDeadline;
+
+        USDC = _usdc;
+    }
+
+    /************************************************
+     *  SETTERS
+     ***********************************************/
+
+    /**
+     * @notice sets the floor and cap of a particular asset notional
+     * @dev can only be called by owner
+     * @param _underlying underlying asset address
+     * @param _min minimum notional value allowed in USD with 6 decimals
+     * @param _max maximum notional value allowed in USD with 6 decimals
+     */
+    function setMinMaxNotional(
+        address _underlying,
+        uint256 _min,
+        uint256 _max
+    ) external onlyOwner {
+        require(_underlying != address(0), "OTCWrapper: asset address cannot be 0");
+        require(_min > 0, "OTCWrapper: minimum notional cannot be 0");
+        require(_max > 0, "OTCWrapper: maximum notional cannot be 0");
+
+        minMaxNotional[_underlying].min = _min;
```

Comment:
better to do `minMaxNotional[_underlying] = MinMaxNotional(_min, _max)`

Reply:
https://github.com/ribbon-finance/GammaProtocol-OTC/pull/2/commits/498221674f9fe002c5d5472d4268ee22e0e76b0d